utils/load_path.ml

```diff
@@ -109,14 +109,14 @@ let is_basename fn = Filename.basename fn = fn
 
 let find fn =
   assert (not Config.merlin || Local_store.is_bound ());
-  if is_basename fn then
+  if is_basename fn && not !Sys.interactive then
     SMap.find fn !files
   else
     Misc.find_in_path (get_paths ()) fn
```

Comment:
FWIW in https://github.com/ocaml/ocaml/issues/10087#issuecomment-745204175 I was only envisaging the fractionally longer:
```suggestion
  if is_basename fn then
    try SMap.find fn !files
    with Not_found when !Sys.interactive -> Misc.find_in_path (get_paths ()) fn
  else
    Misc.find_in_path (get_paths ()) fn
```
(i.e. not worrying about updating the clearly out-of-date cache)

Reply:
The two versions might return different results if `fn` was already present and the user dropped a new version in a directory with a higher precedence.

Reply:
That's an extremely good point :facepalm: